pkg/controllers/cluster/import.go

```diff
@@ -139,6 +140,21 @@ func (i *importHandler) deleteOldAgent(cluster *fleet.Cluster, kc kubernetes.Int
 		}
 	}
 
+	if i.systemNamespace != config.DefaultNamespace {
+		// Attempt to delete the agent in the default namespace if it has changed.
+		if err := kc.AppsV1().Deployments(config.DefaultNamespace).Delete(i.ctx, config.AgentConfigName, metav1.DeleteOptions{}); err != nil {
+			return err
+		}
+		apply, err := apply.NewForConfig(restConfig)
```

Comment:
Actually we might just need to drop this. I doubt this will actually work and magically clean up everything.

Reply:
Agreed. Dropped